plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/snowflake/SnowflakeIcebergTableOperationsProvider.java

```diff
@@ -13,32 +13,39 @@
  */
 package io.trino.plugin.iceberg.catalog.snowflake;
 
+import com.google.common.collect.ImmutableMap;
 import com.google.inject.Inject;
 import io.trino.filesystem.TrinoFileSystemFactory;
+import io.trino.plugin.iceberg.IcebergConfig;
 import io.trino.plugin.iceberg.catalog.IcebergTableOperations;
 import io.trino.plugin.iceberg.catalog.IcebergTableOperationsProvider;
 import io.trino.plugin.iceberg.catalog.TrinoCatalog;
 import io.trino.plugin.iceberg.fileio.ForwardingFileIo;
 import io.trino.spi.connector.ConnectorSession;
 import org.apache.iceberg.snowflake.SnowflakeIcebergTableOperations;
 
+import java.util.HashMap;
 import java.util.Optional;
 
+import static io.trino.plugin.iceberg.IcebergUtil.loadManifestCachingProperties;
 import static java.util.Objects.requireNonNull;
 
 public class SnowflakeIcebergTableOperationsProvider
         implements IcebergTableOperationsProvider
 {
     private final String snowflakeDatabase;
     private final TrinoFileSystemFactory fileSystemFactory;
+    private final IcebergConfig icebergConfig;
```

Comment:
IcebergConfig is a mutable. we avoid shared mutable state. please do not keep the config on the field, just extract what you need from it in the constructor.